### PR TITLE
Make coreclr_property_bag_t handle lifetime of property strings

### PR DESCRIPTION
**Pull request by elinor-fung**
- Make coreclr_property_bag_t handle lifetime of property strings
- Basic e2e test for runtime config properties

### CODE REVIEW

**Comment by elinor-fung** [comment]
This is in anticipation of being able to separate calculating properties/initializing and actually running

File: src/corehost/cli/hostpolicy/hostpolicy.cpp [hunk 1]
```diff
@@ -569,28 +554,10 @@ SHARED_API int corehost_main_with_output_buffer(const int argc, const pal::char_
...
-        trace_hostpolicy_entrypoint_invocation(location);
-        trace::info(_X("}"));
-
-        trace::info(_X("Deps file: %s"), hostpolicy_init.deps_file.c_str());
-        for (const auto& probe : hostpolicy_init.probe_paths)
-        {
-            trace::info(_X("Additional probe dir: %s"), probe.c_str());
-        }
-    }
-
```
Comment (sdmaclea) [inline src/corehost/cli/hostpolicy/hostpolicy.cpp]: Where did the tracing code move to?  I didn't see it being inserted into another function.  (Here and below)

Comment (elinor-fung) [reply src/corehost/cli/hostpolicy/hostpolicy.cpp]: It's in `corehost_init`. I basically split `corehost_main_init` out into that, so the diff doesn't show it as added/changed - I think that section of code is actually collapsed in the default diff view =/
In the changed file, it is at lines 460-475 and line 482.

File: src/test/HostActivationTests/RuntimeProperties.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should().Pass()
+                .And.HaveStdErrContaining($"Property {sharedState.AppTestPropertyName} = {sharedState.AppTestPropertyValue}")
+                .And.HaveStdOutContaining($"AppContext.GetData({sharedState.AppTestPropertyName}) = {sharedState.AppTestPropertyValue}");
+        }
+
+        [Fact]
+        public void DuplicateConfigProperty_AppConfigValueUsed()
```
Comment (vitek-karas) [inline src/test/HostActivationTests/RuntimeProperties.cs]: Add one more test which just reads the property from the framework runtime config - for completeness.

**Pull request merged**